cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -100,19 +109,68 @@ pub enum Error {
     Network(#[from] network::Error),
     #[error(transparent)]
     Wasm(#[from] wasm::Error),
+    #[error("alias \"{alias}\" already exist. Use --force to override it.")]
+    AliasAlreadyExist { alias: String },
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
+        self.validate_alias()?;
+
         let res = self.run_against_rpc_server(None, None).await?.to_envelope();
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
+                self.save_contract_id(&contract);
                 println!("{contract}");
             }
         }
         Ok(())
     }
+
+    fn validate_alias(&self) -> Result<(), Error> {
+        let alias = self.alias();
+
+        if alias.is_empty() {
+            return Ok(());
+        }
+
+        let path = self.alias_path();
+
+        if path.exists() && !self.force {
+            Err(Error::AliasAlreadyExist { alias })
+        } else {
+            Ok(())
+        }
+    }
+
+    fn alias(&self) -> String {
+        match &self.alias {
+            Some(n) => n.to_string(),
+            None => String::new(),
+        }
```

Comment:
Just for your info.
```suggestion
        self.alias.as_ref().map(Clone::clone).unwrap_or_default()
 ```
 
 However, I suggest removing method.